climakitae/util/utils.py

```diff
@@ -946,18 +946,26 @@ def add_dummy_time_to_wl(wl_da: xr.DataArray) -> xr.DataArray:
     # Creating map from frequency name to freq var needed for pandas date range
     if wl_time_dim == "time_delta":
         time_freq_name = wl_da.frequency
-        name_to_freq = {"hourly": "h", "daily": "D", "monthly": "ME"}
+        name_to_freq = {"hourly": "h", "daily": "D", "monthly": "MS"}
     else:
         time_freq_name = wl_time_dim.split("_")[0]
-        name_to_freq = {"hours": "h", "days": "D", "months": "ME"}
+        name_to_freq = {"hours": "h", "days": "D", "months": "MS"}
 
     # Creating dummy timestamps
     timestamps = pd.date_range(
         "2000-01-01",
-        periods=len(wl_da[wl_time_dim]),
+        periods=int(
+            len(wl_da[wl_time_dim]) * 1.1
```

Comment:
Note: this solution is made (` * 1.1`) because I didn't want to change the datetime data type away from a pandas datetime object. I know that this could be a cftime.DatetimeNoLeap object, but I am worried about what might break because of this subtle change (i.e. pandas datetimes support .dt accessor, but cftime.DatetimeNoLeap doesn't).

Reply:
This not pretty but it will definitely keep the data type the same.

Reply:
I actually don't understand why you are multiplying by 1.1, even with your comments. Can you elaborate on why you do this? Is it to ensure you capture leap days?

Reply:
I may try to do this instead actually, let me see how hard it is. I just put 1.1 for convenience.